opm/autodiff/BlackoilModelEbos.hpp

```diff
@@ -298,17 +304,21 @@ namespace Opm {
                 updateState(x,reservoir_state);
                 wellModel().updateWellState(xw, well_state);
 
-                // since the solution was changed, the cache for the intensive quantities
-                // are invalid
-                ebosSimulator_.model().invalidateIntensiveQuantitiesCache(/*timeIdx=*/0);
+                // since the solution was changed, the cache for the intensive quantities are invalid
+                // ebosSimulator_.model().invalidateIntensiveQuantitiesCache(/*timeIdx=*/0);
+            }
 
-                // solver has succeed i.e. no need for restart.
-                isRestart_ = false;
+            if( converged && (iteration >= nonlinear_solver.minIter()) )
+            {
+                // in case of convergence we do not need to reset intensive quantities
```

Comment:
You can not reuse the intensive quantities if the solver fails because you will need to restart the simulation from the solution from the last timestep.  (not just the last iteration)